sim/midas/src/main/cc/simif_emul.cc

```diff
@@ -15,8 +15,10 @@
 #include "bridges/cpu_managed_stream.h"
 
 uint64_t main_time = 0;
-std::unique_ptr<mmio_t> master;
-std::unique_ptr<mmio_t> dma;
+
+mmio_t *simif_emul_t::master = nullptr;
+mmio_t *simif_emul_t::dma = nullptr;
+mm_t *simif_emul_t::slave[MEM_NUM_CHANNELS] = {nullptr};
```

Comment:
I know you can initialize `master` and `dma` to something real here and remove the new and delete calls for them in the constructor/destructor.  That would make the memory handling thread safe because they will be allocated before `mainI()` is called.  There's more logic for handling the `slave` channels, so I don't think that will work here.